pkg/noderesourcetopology/filter_test.go

```diff
@@ -598,10 +790,40 @@ func makeMultiContainerPodWithResourceList(name string, resourcesList []v1.Resou
 	}
 }
 
+func makeMultiInitContainerAndContainerPodWithResourceList(name string, initResourceList, resourcesList []v1.ResourceList) *v1.Pod {
+	p := makeMultiContainerPodWithResourceList(name, initResourceList)
+	pod := makeMultiContainerPodWithResourceList(name, resourcesList)
+	pod.Spec.InitContainers = p.Spec.Containers
+
+	return pod
+}
+
 func cloneResourceList(rl v1.ResourceList) v1.ResourceList {
 	res := make(v1.ResourceList)
 	for name, qty := range rl {
 		res[name] = qty
 	}
 	return res
 }
+
+// makeResList intention is to decrease the amount of repetitive code and improve test writing speed
+// it gets a variable amount of strings, where each represent a quantity of different resource in the following order
+// [0] = v1.ResourceCPU
+// [1] = v1.ResourceMemory
+// [2] =  hugepages2Mi
+// [3] = nicResourceName
+func makeResList(quan ...string) v1.ResourceList {
```

Comment:
the intent here is good, but this helper now makes more hidden (not more explicit) which resources has which quantity. In the end it seems to be saving this
```
resourceName = resource.MustParse(q)
```
which doesn't look like a big improvement.
Additionally, having the tests very explicit and all the context laid down explicitely, having at the cost of some repetition, is IMO a very good thing in the test flow.

Reply:
I was trying to have more "matrix-like" tests to ease the test writing process, so all the developer needs to do is change some numbers, and avoid copy-paste big chunks of code which sometimes might lead to mistakes.

Reply:
the intent is good but I don't think this specific implementation is on target

Reply:
Revert it. Done